bumble/device.py

```diff
@@ -993,7 +993,7 @@ def __init__(
                 for descriptor in characteristic.get("descriptors", []):
                     new_descriptor = Descriptor(
                         attribute_type=descriptor["descriptor_type"],
-                        permissions=descriptor["permission"],
+                        permissions=int(descriptor["permission"], 0),
```

Comment:
Shouldn't the JSON property be an int? Allowing a string where an int is expected is probably setting the wrong expectation. If we want to allow permissions to be a string, we could allow passing an array of constant names and parse the strings into ints.

Reply:
did a quick experiment:
```
>>> type(json.loads('{"x":9}')["x"])
<type 'int'>
>>> type(json.loads('{"x":"9"}')["x"])
<type 'unicode'>
>>> type(json.loads('{"x":"0x10"}')["x"])
<type 'unicode'>
```

Overall that makes sense. I was setting permission to "0x01" as a way to set the characteristics as readable, which ironically isn't very readable. 

proper fix incoming